Inductor comms reorder logs to tlparse

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #155737

Hacked test_inductor_collectives test to demonstrate this works:
https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/whc/de50ff33-f460-406b-bfa9-457e6e17395b/custom/-_0_0_0/reorder_communication_preserving_peak_memory_9.txt?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=10000

Follow up: it would be nice to move the logging out of this pass and
into the broader comms pass loop, where the before/after each pass
visualization could be logged into the same tlparse file.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

wconstab committed:
Inductor comms reorder logs to tlparse

Hacked test_inductor_collectives test to demonstrate this works:
https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/whc/de50ff33-f460-406b-bfa9-457e6e17395b/custom/-_0_0_0/reorder_communication_preserving_peak_memory_9.txt?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=10000

Follow up: it would be nice to move the logging out of this pass and
into the broader comms pass loop, where the before/after each pass
visualization could be logged into the same tlparse file.

[ghstack-poisoned]

bdhirsh commented on torch/_inductor/comms.py:
```diff
@@ -112,42 +113,52 @@ def reorder_communication_preserving_peak_memory(
-        overlap_log.info(
-            "Please `pip install tabulate` to nicely render overlap stats."
+        reorder_log_str += (
+            "Please `pip install tabulate` to nicely render overlap stats.\n"
```
side thought - I have no idea if internal jobs generally come with `tabulate` installed and/or if this is easy to change from the trainer side, but should the default be to print a slightly uglier string rather than to print nothing at all?

wconstab replied:
the default is actually an ugly string. i think you just missed it (2 lines below).  I also tested that this works / shows up in tlparse properly.  And the output can even be copypasted into python if you want to reformat it with tabulate after the fact.

bdhirsh reviewed (approved):
nice!

wconstab committed:
Update on "Inductor comms reorder logs to tlparse"


Hacked test_inductor_collectives test to demonstrate this works:
https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/whc/de50ff33-f460-406b-bfa9-457e6e17395b/custom/-_0_0_0/reorder_communication_preserving_peak_memory_9.txt?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=10000

Follow up: it would be nice to move the logging out of this pass and
into the broader comms pass loop, where the before/after each pass
visualization could be logged into the same tlparse file.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

wconstab commented:
@pytorchbot merge -i
